core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4MultiQueriesPreparedStatement.java

```diff
@@ -0,0 +1,393 @@
+            connection.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "<init>", "", p);
+        }
+
+    }
+
+    TrafT4MultiQueriesPreparedStatement(TrafT4Connection connection, String sql, int resultSetType,
+            int resultSetConcurrency, int resultSetHoldability, String stmtLabel) throws SQLException {
+        super(connection, sql, resultSetType, resultSetConcurrency, resultSetHoldability, stmtLabel);
+        if (connection.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection.props_, connection, sql, resultSetType,
+                    resultSetConcurrency, resultSetHoldability, stmtLabel);
+            connection.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "<init>", "", p);
+        }
+
+        if (resultSetType != ResultSet.TYPE_FORWARD_ONLY && resultSetType != ResultSet.TYPE_SCROLL_INSENSITIVE
+                && resultSetType != ResultSet.TYPE_SCROLL_SENSITIVE) {
+            throw TrafT4Messages.createSQLException(connection_.props_, connection_.getLocale(),
+                    "invalid_resultset_type", null);
+        }
+        if (resultSetConcurrency != ResultSet.CONCUR_READ_ONLY && resultSetConcurrency != ResultSet.CONCUR_UPDATABLE) {
+            throw TrafT4Messages.createSQLException(connection_.props_, connection_.getLocale(),
+                    "invalid_resultset_concurrency", null);
+        }
+        if ((resultSetHoldability != 0) && (resultSetHoldability != ResultSet.CLOSE_CURSORS_AT_COMMIT)
+                && (resultSetHoldability != ResultSet.HOLD_CURSORS_OVER_COMMIT)) {
+            throw TrafT4Messages.createSQLException(connection_.props_, connection_.getLocale(), "invalid_holdability",
+                    null);
+        }
+
+        sqlArr = sql.split(";");
+        pstmtArr = new TrafT4PreparedStatement[sqlArr.length];
+        paramDescs = new int[sqlArr.length][3];
+        int total = 0;
+        int currentParamsLen = 0;
+        for (int i = 0; i < sqlArr.length; i++) {
+            pstmtArr[i] = new TrafT4PreparedStatement(connection, sqlArr[i], resultSetType, resultSetConcurrency,
+                    resultSetHoldability, stmtLabel);
+            pstmtArr[i].prepare(sqlArr[i], queryTimeout_, resultSetHoldability_);
+
+            // this is to setup the paramDescs, each paramDesc has 3 column,
+            // 1: index of preparedStatement
+            // 2: num of params for current pstmt
+            // 3: accumulated params for current index
+            // this varible is used when there invoking setXXX api
+            if (pstmtArr[i].inputDesc_ != null) {
+                currentParamsLen = pstmtArr[i].inputDesc_.length;
+                total += currentParamsLen;
+            }
+            int[] paramDesc = { i, currentParamsLen, total };
+            paramDescs[i] = paramDesc;
+
+        }
+    }
+
+    public boolean execute(String sql) throws SQLException {
```

Comment:
This method is invalid in PreparedStatement.  A preparedStatement can't execute different SQL.

Reply:
The benefit of preparedstatements is to prepare the query once and to execute multiple times. By breaking each semicolon separated sql into its own preparestatement, are we losing out on this feature?

Would it be more appropriate to implement this feature for dynamic statements "Statement" as opposed to "PreparedStatement" ?

Reply:
@selvaganesang 
i am not sure whether user will do execute(sql) in a preparedStmt. but this api is valid in JKD, it's inherted from java.sql.statement, if it's invalid in trafodion, i can remove it.

@venkat1m there may have possible that user want to exec "insert into t values (?); insert into t2 values(?,?)" in one preparedstatement to avoid sql code insertion attacks, in this PR, it will separate to 2 preparedStmts, and each one can execute multi-times.  so the prepare is needed.

Reply:
It could be valid from interface inheritance point of view. However, the JDBC specification  clearly says the following

boolean execute(String sql)
                throws SQLException
Note:This method cannot be called on a PreparedStatement or CallableStatement.

It is not a Trafodion restriction. It is against the intent of PreparedStatement functionality as per the specification, 

All other flavors of execute call in Statement object can't be called in PreparedStatement or CallableStatement.

Reply:
@selvaganesang so your suggestion is  to remove the method or  throw exception when user invoke the method?
i see TrafT4PreparedStatement not mark the methods( execute(sql) ) as not support